git_glossary.md

```diff
@@ -0,0 +1,31 @@
+# Git Glossary 
+
+This git glossary contains definitions of terms used in this repository.
+
+There is another dcoument that describes a mental model of git that
+when understood can help one master git concepts at a deeper level. 
+
+## commit
+As a noun: A single point in history of a repository. The entire history of a project is represented as a set
+of interrelated commits. Each commit is uniquely identified by a 40-character hexadecimal string which is created using
+a cryptographic hash function, also referred to has the SHA of the commit. Sometimes the SHA is abbreviated to the 
+first 7 characters of the 40-character string.
+
+As a verb: The action of storing a new snapshot of the project's state in the Git history, by creating a new commit. 
+
+## commit object
+An object which contains the information about a particular commit. 
+
+## object database
+
+## repository
+A collection of [references](#references) to commit objects and an [object database](#object database) of commit objects. 
```

Comment:
`[object database](#object database)` - for this to work (and strangely to be rendered correctly as well) we need to use a hyphen between object and database (even though there is a space in the header that we are referencing): `[object database](#object-database)`

Reply:
Done